iwyu.cc

```diff
@@ -3691,8 +3691,12 @@ class IwyuAstConsumer
     num_edits += preprocessor_info().FileInfoFor(main_file)
         ->CalculateAndReportIwyuViolations();
 
-    // We need to force the compile to fail so we can re-run.
-    exit(EXIT_SUCCESS_OFFSET + num_edits);
+    // We need to force the compile to fail so we can re-run,
+    // unless --exit_code is specified
+    if (GlobalFlags().exit_code && num_edits == 0)
+      exit(EXIT_SUCCESSFULLY);
+    else
+      exit(EXIT_SUCCESS_OFFSET + num_edits);
```

Comment:
Even though `exit()` takes an `int`, functions like `wait()` and `waitpid()` only guarantee that you can get the bottom eight bits. `waitid()` has addressed this, but is only rarely used.

This means that if `EXIT_SUCCESS_OFFSET + num_edits % 256 == 0`, most shells/launchers may interpret the result as success. :-)

Reply:
Yeah, I pointed that out in my PR description. It's an existing bug, I just moved the existing `exit(EXIT_SUCCESS_OFFSET + num_edits)` into an `else`. I decided to not touch it in order to avoid changing the behavior of the code when running without `--exit_code`, but that might have been the wrong choice.

Reply:
Whoops! Sorry for not reading the description properly. Great work!

Reply:
This should be resolved by https://github.com/include-what-you-use/include-what-you-use/commit/ef76947aa66be44473037cf45e5f34d01019a412